src/google-images.coffee

```
@@ -30,17 +34,52 @@ module.exports = (robot) ->
 imageMe = (msg, query, animated, faces, cb) ->
   cb = animated if typeof animated == 'function'
   cb = faces if typeof faces == 'function'
-  q = v: '1.0', rsz: '8', q: query, safe: 'active'
-  q.imgtype = 'animated' if typeof animated is 'boolean' and animated is true
-  q.imgtype = 'face' if typeof faces is 'boolean' and faces is true
-  msg.http('http://ajax.googleapis.com/ajax/services/search/images')
-    .query(q)
-    .get() (err, res, body) ->
-      images = JSON.parse(body)
-      images = images.responseData?.results
-      if images?.length > 0
-        image = msg.random images
-        cb ensureImageExtension image.unescapedUrl
+  googleCseId = process.env.HUBOT_GOOGLE_CSE_ID
+  if googleCseId
+    # Using Google Custom Search API
+    googleApiKey = process.env.HUBOT_GOOGLE_API_KEY
+    if !googleApiKey
+      msg.robot.logger.error "Missing Google API key"
```

`logger.error` isn't visible to chat users, so the command would silently fail. It's usually better to `msg.reply` about the missing environment variable.